lib/model/model.go

```diff
@@ -1569,9 +1569,13 @@ func (m *Model) updateLocalsFromScanning(folder string, fs []protocol.FileInfo)
 	runner, _ := m.folderRunners[folder]
 	m.fmut.RUnlock()
 
-	fs = runner.validateLocalChanges(fs)
-
-	m.updateLocals(folder, fs)
+	if runner == nil {
+		l.Infoln("bug: folderrunner is nil (should only happen in tests)")
+		m.updateLocals(folder, fs)
+	} else {
+		// validate all local changes and update the database
+		fs = runner.validateAndUpdateLocalChanges(fs)
+	}
```

Comment:
Unfortunately we do need this. Test cases in model_test.go fail, since they never start the folder and thus runner will be nil.